### PR TITLE
[Feature] add Draggable pip for local user video stream window

### PR DESCRIPTION
**Pull request by lpeixin**
## Purpose
make the local user video window being draggable

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Join a call with remote participants try to drag the local video window pip to verify if it can be draggable and won't be outside of parent view boundary. Also, we can verify the pip not to cover the header bar or bottom drawer. Then test all these in landscape mode.

### CODE REVIEW

**Review by Mohtasim — DISMISSED** [review]

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/DragTouchListener.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+class DragTouchListener : View.OnTouchListener {
+
+    private var curX = 0f
+    private var curY = 0f
+    private var startX = 0f
+    private var startY = 0f
+    private var tranX = 0f
+    private var tranY = 0f
+    private lateinit var view: View
+    private var listener: OnDragListener? = null
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/DragTouchListener.kt:18]: Where is this being set?

File: azure-communication-ui/azure-communication-ui/src/calling/res/layout-land/azure_communication_ui_calling_call_local_pip_video.xml [hunk 2]
```diff
@@ -19,17 +19,25 @@
...
         android:layout_width="0dp"
         android:layout_height="0dp"
+        android:id="@+id/azure_communication_ui_call_local_pip_wrapper"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintDimensionRatio="16:9"
+        app:layout_constraintStart_toStartOf="@+id/azure_communication_ui_guideline"
+        app:layout_constraintEnd_toEndOf="parent"
+        >
+
+    <LinearLayout
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/calling/res/layout-land/azure_communication_ui_calling_call_local_pip_video.xml:32]: Can we indent these child views

**Review by phileo99 — CHANGES_REQUESTED** [review]

File: azure-communication-ui/azure-communication-ui/src/calling/res/layout-land/azure_communication_ui_calling_call_local_pip_video.xml (hunk 2 above)
Comment (lpx1989) [reply azure-communication-ui/azure-communication-ui/src/calling/res/layout-land/azure_communication_ui_calling_call_local_pip_video.xml]: sure, updated now.

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/DragTouchListener.kt (hunk 1 above)
Comment (lpx1989) [reply azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/DragTouchListener.kt]: thx for pointing this, I cleaned up the unused listener now.

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/DragTouchListener.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    private var curX = 0f
+    private var curY = 0f
+    private var startX = 0f
+    private var startY = 0f
+    private var tranX = 0f
+    private var tranY = 0f
+    private lateinit var view: View
+
+    internal constructor() {}
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/DragTouchListener.kt:19]: can we convert this to primary constructor

**Review by phileo99 — CHANGES_REQUESTED** [review]

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/DragTouchListener.kt (hunk 3 above)
Comment (lpx1989) [reply azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/DragTouchListener.kt]: addressed now.

**Review by phileo99 — APPROVED** [review]
tested, looks good
👍

**Review by Mohtasim — APPROVED** [review]

**Pull request merged**